libmisc/salt.c

```diff
@@ -413,7 +645,7 @@ static /*@observer@*/const char *gensalt (size_t salt_size)
 #ifdef USE_YESCRYPT
 	} else if (0 == strcmp (method, "YESCRYPT")) {
 		MAGNUM(result, 'y');
-		strcat(result, YESCRYPT_salt_cost((int *)arg));
+		strcat(result, YESCRYPT_salt_cost((long *)arg));
```

Comment:
Looks like this was actually a bug being fixed here. So at least this bug fix should get in even if this PR maybe doesn't.

Reply:
Indeed!

Reply:
I've opened https://github.com/shadow-maint/shadow/pull/358 to keep this in mind, if the current PR doesn't get merged.